src/pages/index.js

```diff
@@ -68,77 +120,136 @@ const editModalPopup = new PopupWithForm(
+  avatarSelector: ".profile__image",
 });
 
-editModalCloseButton.addEventListener("click", () => {
-  editModalPopup.close();
-});
-
+// Function to handle edit profile form submission
 function editProfileModal() {
   const newName = editModalTitleInput.value;
   const newJob = editModalSubtitleInput.value;
-  userInfo.setUserInfo({ name: newName, job: newJob });
-  editModalPopup.close();
-}
 
+  editModalPopup.renderLoading(true);
+
+  api
+    .updateProfile()
+    .then(() => {
+      console.log("Profile updated successfully:");
+      userInfo.setUserInfo({ name: newName, job: newJob });
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    })
+    .finally(() => {
+      editModalPopup.renderLoading(false);
+      editModalPopup.close();
+    });
+}
+// Selecting profile form elements
 const profileAddedForm = document.forms["add-form"];
-
 const profileForm = document.forms["modal-form"];
 
+// Creating form validators
 const cardFormValidator = new FormValidator(config, profileAddedForm);
 const profileFormValidator = new FormValidator(config, profileForm);
 
+// Disabling form buttons
 cardFormValidator.disableButton();
 profileFormValidator.disableButton();
 
+// Enabling form validation
 cardFormValidator.enableValidation();
 profileFormValidator.enableValidation();
 
+// Function to handle image click
 function handleImageClick(cardData) {
   imagePopup.open(cardData);
 }
 
+// Function to create card HTML
 function createCard(item) {
-  const card = new Card(item, "#card-template", handleImageClick);
-  return card.generateCard();
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButtonClick,
+    handleLikeButtonClick,
+    handleAvatarClick
+  );
+
+  return cardElement.getView();
 }
 
+// Creating instance of PopupWithImage
 const imagePopup = new PopupWithImage("#modal-image-preview");
-
 imagePopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  ".cards__list"
-);
-
-cardSection.renderItems();
-
+// Function to render a single card
 function renderCard(cardData) {
+  const cardSection = new Section(
+    {
+      items: cardData,
+      renderer: createCard,
+    },
+    ".cards__list"
+  );
   const cardElement = createCard(cardData);
   cardSection.addItem(cardElement);
 }
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d9d73c7f-57bf-4e5e-8837-ef7b15edaabc",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    const cardSection = new Section(
+      {
+        items: cards,
+        renderer: createCard,
+      },
+      ".cards__list"
+    );
+
+    cardSection.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error fetching initial cards:", error);
+  });
+
+api
+  .getUserInfo()
```

Comment:
Better to use `Promise.all` to load initial data

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all